File: versioned_docs/version-1.20.x/examples/go/src/lifecycle_policy_browse.go
package main

import (
	"context"

	reduct "github.com/reductstore/reduct-go"
)

func main() {
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{APIToken: "my-token"})

	lifecycles, err := client.GetLifecycles(context.Background())
	if err != nil {
		panic(err)
	}

	for _, lifecycle := range lifecycles {
		println("Lifecycle:", lifecycle.Name)
		println("Mode:", string(lifecycle.Mode))
		println("Running:", lifecycle.IsRunning)
		println("Provisioned:", lifecycle.IsProvisioned)
	}

	detail, err := client.GetLifecycle(context.Background(), "my-lifecycle")
	if err != nil {
		panic(err)
	}
	println("Lifecycle:", detail.Info.Name)
	println("Bucket:", detail.Settings.Bucket)
}

File: versioned_docs/version-1.20.x/examples/py/src/lifecycle_policy_browse.py
import asyncio

from reduct import Client


async def main():
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        for lifecycle in await client.get_lifecycles():
            print("Lifecycle:", lifecycle.name)
            print("Mode:", lifecycle.mode)
            print("Running:", lifecycle.is_running)
            print("Provisioned:", lifecycle.is_provisioned)

        details = await client.get_lifecycle_detail("my-lifecycle")
        print("Lifecycle:", details.info.name)
        print("Settings:", details.settings)


asyncio.run(main())
